newsroom/push/tasks.py

```diff
@@ -13,33 +12,34 @@
 notifier = NotificationManager()
 
 
-@contextmanager
-def locked(_id: str, service: str):
+def get_lock(_id: str, service: str) -> str | None:
     lock_name = f"notify-{service}-{_id}"
     if not lock(lock_name, expire=300):
         logger.debug(f"Lock conflict on {lock_name}")
-        return
-
-    logger.debug("Starting task %s", lock_name)
-    try:
-        yield lock_name
-    finally:
-        unlock(lock_name)
-        logger.debug("Done with %s", lock_name)
+        return None
+    return lock_name
 
 
 @celery.task
 async def notify_new_wire_item(_id, check_topics=True):
-    with locked(_id, "wire"):
+    lock = get_lock(_id, "wire")
+    if not lock:
+        return
+    try:
         logger.info("Send notifications for wire item %s", _id)
         item = await WireSearchServiceAsync().service.find_by_id(_id)
         if item:
             await notifier.notify_new_item(item.to_dict(), check_topics=check_topics)
+    finally:
+        unlock(lock)
 
 
 @celery.task
 async def notify_new_agenda_item(_id, check_topics=True, is_new=False):
-    with locked(_id, "agenda"):
+    lock = get_lock(_id, "wire")
```

Comment:
I think this should be `agenda`. Also variable name here

```suggestion
    agenda_lock = get_lock(_id, "agenda")
```